### PR TITLE
ROC-2485 Prevent defendant response after CCJ has been requested

### PR DESCRIPTION
**Pull request by jacmurphy**
### JIRA link ###
https://tools.hmcts.net/jira/browse/ROC-2485

### Change description ###
This prevents defendants, who have no previously linked their account to a claim before a CCJ has been requested, to respond to a claim.

### [WIP] ###
Updating yarn tests now.

**Does this PR introduce a breaking change?** (check one with "x")

```
[] Yes
[X] No
```

### CODE REVIEW

File: src/main/features/first-contact/guards/claimExpiredGuard.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import * as express from 'express'
+import { ErrorPaths } from 'first-contact/paths'
+import { User } from 'idam/user'
+
+const logger = require('@hmcts/nodejs-logging').getLogger('first-contact/guards/claimReferenceMatchesGuard')
+
+export class ClaimExpiredGuard {
```
Comment (timja-kainos) [inline src/main/features/first-contact/guards/claimExpiredGuard.ts]: this needs renaming, something like `ClaimantRequestedCCJGuard` or `ClaimantAlreadyRequestedCCJGuard`

File: src/main/features/first-contact/guards/claimExpiredGuard.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import * as express from 'express'
+import { ErrorPaths } from 'first-contact/paths'
+import { User } from 'idam/user'
+
+const logger = require('@hmcts/nodejs-logging').getLogger('first-contact/guards/claimReferenceMatchesGuard')
```
Comment (timja-kainos) [inline src/main/features/first-contact/guards/claimExpiredGuard.ts]: change this `claimReferenceMatchesGuard` to match the name of this file

File: src/main/features/first-contact/guards/claimExpiredGuard.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+export class ClaimExpiredGuard {
+  static async requestHandler (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
+    const user: User = res.locals.user
+    if (user.claim.countyCourtJudgmentRequestedAt !== undefined) {
+      logger.error('Defendant cannot counter claim with a CCJ against them - redirect to request denied page ')
```
Comment (timja-kainos) [inline src/main/features/first-contact/guards/claimExpiredGuard.ts]: a counter claim is something else, how about:
> cannot respond

Also it isn't a request denied page, how about:
> redirecting to handoff page

File: src/main/features/first-contact/paths.ts [hunk 4]
```diff
@@ -8,4 +8,5 @@ export class Paths {
 
 export class ErrorPaths {
   static readonly claimSummaryAccessDeniedPage = new RoutablePath('/first-contact/access-denied')
+  static readonly ccjRequestedAccessDeniedPage = new RoutablePath('/first-contact/claimant-requested-for-ccj')
```
Comment (timja-kainos) [inline src/main/features/first-contact/paths.ts]: > ccjRequestedHandoffPage?

File: src/test/features/first-contact/routes/claim-summary.ts [hunk 5]
```diff
@@ -6,20 +6,21 @@ import { attachDefaultHooks } from '../../../routes/hooks'
 import '../../../routes/expectations'
 import { checkAuthorizationGuards } from './checks/authorization-check'
 
-import { ErrorPaths, Paths } from 'first-contact/paths'
+import { ErrorPaths, Paths as DefendantFirstContactPaths } from 'first-contact/paths'
```
Comment (timja-kainos) [inline src/test/features/first-contact/routes/claim-summary.ts]: why make this change? doesn't look like it adds any benefit

Comment (jacmurphy) [reply src/test/features/first-contact/routes/claim-summary.ts]: I was getting errors with TSLint as i had

File: src/main/features/first-contact/guards/claimExpiredGuard.ts (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/first-contact/guards/claimExpiredGuard.ts]: Updating now

File: src/main/features/first-contact/paths.ts (hunk 4 above)
Comment (jacmurphy) [reply src/main/features/first-contact/paths.ts]: I prefer your suggestion. Updated now

File: src/main/features/first-contact/guards/claimExpiredGuard.ts (hunk 3 above)
Comment (jacmurphy) [reply src/main/features/first-contact/guards/claimExpiredGuard.ts]: Updated

File: src/main/features/first-contact/guards/claimExpiredGuard.ts (hunk 1 above)
Comment (jacmurphy) [reply src/main/features/first-contact/guards/claimExpiredGuard.ts]: Update.
(I went with ClaimantRequestedCCJGuard)

File: src/main/features/first-contact/guards/claimExpiredGuard.ts (hunk 2 above)
Comment (jacmurphy) [reply src/main/features/first-contact/guards/claimExpiredGuard.ts]: Updated.

File: src/main/features/first-contact/guards/ClaimantRequestedCCJGuard.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import * as express from 'express'
+import { ErrorPaths } from 'first-contact/paths'
+import { User } from 'idam/user'
+
+const logger = require('@hmcts/nodejs-logging').getLogger('first-contact/guards/ClaimantRequestedCCJGuard')
```
Comment (timja-kainos) [inline src/main/features/first-contact/guards/ClaimantRequestedCCJGuard.ts]: don't use uppercase for the filename, it causes problems with case sensitive file systems and imports

File: src/test/features/first-contact/routes/claim-summary.ts (hunk 5 above)
Comment (timja-kainos) [reply src/test/features/first-contact/routes/claim-summary.ts]: ErrorPaths, Paths then?

File: src/main/features/first-contact/guards/claimantRequestedCCJGuard.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import * as express from 'express'
+import { ErrorPaths } from 'first-contact/paths'
+import { User } from 'idam/user'
+
+const logger = require('@hmcts/nodejs-logging').getLogger('first-contact/guards/ClaimantRequestedCCJGuard')
```
Comment (timja-kainos) [inline src/main/features/first-contact/guards/claimantRequestedCCJGuard.ts]: logger case doesn't match file name

File: src/main/features/first-contact/paths.ts [hunk 8]
```diff
@@ -8,4 +8,5 @@ export class Paths {
 
 export class ErrorPaths {
   static readonly claimSummaryAccessDeniedPage = new RoutablePath('/first-contact/access-denied')
+  static readonly ccjRequestedHandoffPage = new RoutablePath('/first-contact/claimant-requested-for-ccj')
```
Comment (timja-kainos) [inline src/main/features/first-contact/paths.ts]: this doesn't read nicely to me: `claimant-requested-for-ccj` 
how about:
`claimant-already-requested-ccj`
or
`claimant-has-requested-ccj`

File: src/main/features/first-contact/views/claimant-requested-for-ccj.njk [hunk 9]
```diff
@@ -0,0 +1,14 @@
+{% extends "layout.njk" %}
+
+{% set title = t('CCJ Requested against you') %}
```
Comment (timja-kainos) [inline src/main/features/first-contact/views/claimant-requested-for-ccj.njk]: no need to translate the title and heading

File: src/main/features/first-contact/paths.ts (hunk 8 above)
Comment (jacmurphy) [reply src/main/features/first-contact/paths.ts]: I just took it from your suggestion earlier. Ill update now

Comment (timja-kainos) [reply src/main/features/first-contact/paths.ts]: I meant the page url btw on this one, not the variable

**Pull request merged**